ArmPkg/Library/ArmFfaLib/ArmFfaCommon.h

```diff
@@ -0,0 +1,76 @@
+/** @file
+  Arm FF-A ns common library Header file
+
+  Copyright (c) 2024, Arm Limited. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  @par Glossary:
+     - FF-A - Firmware Framework for Arm A-profile
+     - spmc - Secure Partition Manager Core
+     - spmd - Secure Partition Manager Dispatcher
+
+  @par Reference(s):
+     - Arm Firmware Framework for Arm A-Profile [https://developer.arm.com/documentation/den0077/latest]
+
+**/
+
+#ifndef ARM_FFA_COMMON_LIB_H_
+#define ARM_FFA_COMMON_LIB_H_
+
+/**
+ * Arguments to call FF-A request via SMC/SVC.
+ */
+typedef struct ArmFfaArgs {

```

Comment:
Should we support the x0 - x17 all the way?

Reply:
RIght now. there is no usage using x8 to x17.
And `ArmCallSmc()` and `ArmCallSvc()` implementation in EDK2, only supports from x0 to x7.
So, It would be better to support x0 to x7 right now.

Reply:
hmm, since we are already trying to push through a large PR, it might be worthy to add the changes for `ArmCallSmc` and `ArmCallSvc` than waiting till the needs become pertinent. Please let me know if you think otherwise.

Reply:
But, It would affect other modules which uses `ArmCallSmc()` and `ArmCallSvc()`.
And I think it's another big task. So I think it would be better to separate with this patch series.